### PR TITLE
Chad/#38 authorization

### PR DESCRIPTION
**Pull request by chadlagore**
:hourglass: **Status**: _Ready for review_

:tickets: **Ticket(s)**: #38 

---

## :construction_worker: Changes

+ Added an `authorization` function decorator.
+ Added token generation functionality, passed securely over SSH.
+ Included comments on how to apply it to endpoints.
+ Implemented a trivial endpoint for health checks.
+ Added some testing around auth.

## :flashlight: Testing Instructions

```bash
go test ./cmd -cover
```

### CODE REVIEW

File: cmd/util.go [hunk 1]
```diff
@@ -84,3 +93,61 @@ func removeContents(directory string) error {
...
+	key, err := getGithubKey()
+	if err != nil {
+		return nil, err
+	}
+	return []byte(key.String()), nil
+}
+
+// getGithubKey returns the key generated by 'inertia remote bootstrap [REMOTE]'
+func getGithubKey() (ssh.AuthMethod, error) {
+	pemFile := "/app/host/.ssh/id_rsa_inertia_deploy"
```
Comment (bobheadxi) [inline cmd/util.go]: A bit hesitant about putting this function in `util` since that path (`/app/host/...`) applies specifically to the daemon container

Comment (chadlagore) [reply cmd/util.go]: Mmmm. How about we make this function more general? `getKeyFromPem(pemFile string)`. Then we could possibly unit test it too.

Comment (chadlagore) [reply cmd/util.go]: Actually I think [this](https://github.com/ubclaunchpad/inertia/blob/master/cmd/init.go#L212) is the "go" way to write testable code that does file IO. Something like `func getKeyFromPem(pemFile io.Reader)`

Comment (chadlagore) [reply cmd/util.go]: I can move it back for the time being :+1: Something to think about.

File: cmd/util.go [hunk 2]
```diff
@@ -84,3 +93,61 @@ func removeContents(directory string) error {
...
+// sending HTTP requests to the daemon server.
+func generateToken(key []byte) (string, error) {
+	// No claims for now.
+	return jwt.New(jwt.SigningMethodHS256).SignedString(key)
+}
+
+// getAPIPrivateKey returns the private RSA key to authenticate HTTP
+// requests sent to the daemon. For now, we simply use the GitHub
+// deploy key.
+func getAPIPrivateKey(*jwt.Token) (interface{}, error) {
```
Comment (bobheadxi) [inline cmd/util.go]: @chadlagore why does this return `interface{}`?

**Pull request merged**